### PR TITLE
Add register persistent worker latency metrics  #3671

### PR DESCRIPTION
**Pull request by dominicqi**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [+] New feature provided
- [ ] Improve performance

- Related issues
#3671 
___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.
Add metrics for register persistent worker  ,and Grafana dashboard.

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java [hunk 1]
```diff
@@ -80,39 +91,43 @@ private void onWork(RegisterSource registerSource) {
...
-                } catch (Throwable t) {
-                    logger.error(t.getMessage(), t);
-                }
-            });
-            sources.clear();
+                });
+                sources.clear();
+            }
+        } catch (IOException e) {
+            logger.error("not except exception happen", e);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java]: Why need this log and catch? And clearly, the sentence is not right.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java [hunk 2]
```diff
@@ -80,39 +91,43 @@ private void onWork(RegisterSource registerSource) {
...
-                        }
-                    } else {
-                        int sequence;
-                        if ((sequence = registerLockDAO.getId(scopeId, source)) != Const.NONE) {
-                            try {
-                                dbSource = registerDAO.get(modelName, source.id());
-                                if (Objects.nonNull(dbSource)) {
-                                    if (dbSource.combine(source)) {
-                                        registerDAO.forceUpdate(modelName, dbSource);
+        try (HistogramMetrics.Timer ignored = histogram.createTimer()) {
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java]: Why timer called ignored?

File: docs/en/setup/backend/telemetry/trace-mode-grafana.json [hunk 3]
```diff
@@ -1158,14 +1158,112 @@
         "align": false,
         "alignLevel": null
       }
+    },
```
Comment (wu-sheng) [inline docs/en/setup/backend/telemetry/trace-mode-grafana.json]: You just change the trace-mode json, miss the mesh-mode one.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java (hunk 1 above)
Comment (dominicqi) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java]: i want use the try-with-resource sugar . The Timer#close() throws IOException that  never throw. I will remove the IOException from the close method.

**Comment by wu-sheng** [comment]
@JaredTan95 Is this good?

**Comment by wu-sheng** [comment]
@kezhenxu94 This works as expected. 3 plugin test workload finished in 2m, only source download and compare. I just noticed one place needs to be improved, we can't see the cooperation log.

**Comment by wu-sheng** [comment]
@kezhenxu94 This seems to be the whole log I got, https://builds.apache.org/blue/rest/organizations/jenkins/pipelines/skywalking-agent-test/runs/779/nodes/27/log/?start=0

Do I miss anything?

**Comment by JaredTan95** [comment]
image

image

Looks Good to me.

**Pull request merged**